pkg/pillar/types/zedmanagertypes.go

```diff
@@ -276,28 +279,169 @@ type StorageStatus struct {
 }
 
 // UpdateFromStorageConfig sets up StorageStatus based on StorageConfig struct
-func (ss *StorageStatus) UpdateFromStorageConfig(sc StorageConfig) {
-	ss.DatastoreID = sc.DatastoreID
-	ss.ImageID = sc.ImageID
-	ss.Name = sc.Name
-	ss.NameIsURL = sc.NameIsURL
-	ss.ImageSha256 = sc.ImageSha256
-	ss.Size = sc.Size
-	ss.CertificateChain = sc.CertificateChain
-	ss.ImageSignature = sc.ImageSignature
-	ss.SignatureKey = sc.SignatureKey
-	ss.ReadOnly = sc.ReadOnly
-	ss.Preserve = sc.Preserve
-	ss.Format = sc.Format
-	ss.Maxsizebytes = sc.Maxsizebytes
-	ss.Devtype = sc.Devtype
-	ss.Target = sc.Target
-	if ss.Format == "container" {
-		ss.IsContainer = true
+func (ssPtr *StorageStatus) UpdateFromStorageConfig(sc StorageConfig) {
+	ssPtr.DatastoreID = sc.DatastoreID
+	ssPtr.Name = sc.Name
+	ssPtr.NameIsURL = sc.NameIsURL
+	ssPtr.ImageSha256 = sc.ImageSha256
+	ssPtr.Size = sc.Size
+	ssPtr.CertificateChain = sc.CertificateChain
+	ssPtr.ImageSignature = sc.ImageSignature
+	ssPtr.SignatureKey = sc.SignatureKey
+	ssPtr.ReadOnly = sc.ReadOnly
+	ssPtr.Preserve = sc.Preserve
+	ssPtr.Format = sc.Format
+	ssPtr.Maxsizebytes = sc.Maxsizebytes
+	ssPtr.Devtype = sc.Devtype
+	ssPtr.Target = sc.Target
+	if ssPtr.Format == "container" {
+		ssPtr.IsContainer = true
 	}
 	return
 }
 
+// IsCertsAvailable checks certificate requirement/availability for a storage object
+func (ssPtr *StorageStatus) IsCertsAvailable(safename string) (bool, error) {
+	if !ssPtr.needsCerts() {
+		log.Debugf("%s, Certs are not required\n", safename)
+		return false, nil
+	}
+	cidx, err := ssPtr.getCertCount(safename)
+	return cidx != 0, err
+}
+
+// HandleCertStatus gets the CertObject Status for the storage object
+// True, the certificates are ready
+// False, Certificates are not ready or, there are some errors
+// For Errors, it sets the Error inside the storage object
+func (ssPtr *StorageStatus) HandleCertStatus(safename string,
+	certObjStatusPtr *CertObjStatus) bool {
+	if ret, errStr, errSrc, errTime := ssPtr.checkCertsStatusForObject(safename,
+		certObjStatusPtr); !ret {
+		log.Infof("%s, Certs are still not ready\n", safename)
+		if errStr != "" {
+			ssPtr.SetErrorInfo(errStr, errSrc, errTime)
+		}
+		return ret
+	}
+	if ret, errStr, errSrc, errTime := ssPtr.checkCertsForObject(); !ret {
+		log.Infof("%s, Certs are still not installed\n", safename)
+		if errStr != "" {
+			ssPtr.SetErrorInfo(errStr, errSrc, errTime)
+		}
+		return ret
+	}
+	return true
+}
+
+// needsCerts whether certificates are required for the Storage Object
+func (ssPtr *StorageStatus) needsCerts() bool {
+	if len(ssPtr.ImageSignature) == 0 {
+		return false
+	}
+	return true
+}
+
+// getCertCount returns the number of certificates for the Storage Object
+// called with valid ImageSignature only
+func (ssPtr *StorageStatus) getCertCount(safename string) (int, error) {
```

Comment:
Since this isn't supposed to modify StorageStatus, please define it as not being a pointer i.e. ss types.StorageStatus.